src/Bridge/Symfony/Bundle/Resources/config/api.xml

```diff
@@ -104,7 +104,7 @@
             <argument>false</argument>
 
             <!-- Run after serializer.normalizer.data_uri but before serializer.normalizer.object -->
-            <tag name="serializer.normalizer" priority="-923" />
+            <tag name="serializer.normalizer" priority="-900" />
```

Comment:
Do you know which normalizer was being called instead?

Reply:
No. How can I check it?

Reply:
It breaks with priority -901

Reply:
I think it must be related to a normalizer from another package you're using.

Reply:
Oh, okay, I understand now. Your resource class implements JsonSerializable?

Reply:
Yes it does!

Reply:
What do you expect to happen in this case? If your class implements JsonSerializable, we should respect your desire to customize the output, no?

Reply:
I expect output to be the same as it it was before 2.4.0-beta2. I use \JsonSerializable in other places of my code, not to customize the API output.

Reply:
Let me rephrase that:

I know what you expect in this case. But how do we know what the user expects when they implement JsonSerializable? I think the current priority is more correct.

Reply:
I think that user at least expects that doing an API request for a single item won't return a collection to him.

Reply:
Maybe we can assume that if dev defines serialization groups and puts them into de/normalization_context, then he wants to use those groups instead of jsonSerialize()?

Reply:
We can't do that. The lower priority means our normalizer is not called if you implement JsonSerializable. That to me is expected behavior. It's not expected for a resource class to implement JsonSerializable.

As for why it returns a collection, I'm not really sure without taking a deeper look. We check for iterable in the CollectionNormalizer.

Reply:
> It's not expected for a resource class to implement JsonSerializable.

This should be mentioned in the docs.